src/main/java/org/opentripplanner/model/transfer/TransferConstraint.java

```diff
@@ -196,6 +199,48 @@ public boolean isMinTransferTimeSet() {
     return minTransferTime != NOT_SET;
   }
 
+  /**
+   * This method is used to calculate the earliest-board-time in a forward search, and the
+   * latest-alight-time in a reverse search for a constrained transfer. If the given transfer
+   * constraint is a regular transfer, we use the given {@code regularTransferEBT} function
+   * to calculate the targetTime.
+   * <p>
+   * In a forward search we ADD({@code timeAddOp}) the transfer time to the given
+   * {@code sourceTransitArrivalTime} to find the earliest-board-time for the target trip.
+   * <p>
+   * In a reverse search we SUBTRACT({@code timeAddOp}) the transfer time from the given
+   * {@code sourceTransitArrivalTime} to find the latest-alight-time for the target trip.
+   */
+  public int calculateTransferTargetTime(
+    int sourceTransitArrivalTime,
+    int transferSlack,
+    IntSupplier calcRegularTransferTargetTime,
+    SearchDirection direction
+  ) {
+    // Ignore slack and walking-time for guaranteed and stay-seated transfers
+    if (isFacilitated()) {
+      return sourceTransitArrivalTime;
+    }
+
+    // Ignore transfer, board and alight slack for min-transfer-time
+    if (isMinTransferTimeSet()) {
+      int minTransferTime = getMinTransferTime() + transferSlack;
+      if (direction.isForward()) {
+        int minTransferBoardTime = sourceTransitArrivalTime + minTransferTime;
+        return OTPFeature.MinimumTransferTimeIsDefinitive.isOn()
+          ? minTransferBoardTime
+          : Math.max(minTransferBoardTime, calcRegularTransferTargetTime.getAsInt());
+      } else {
+        int minTransferBoardTime = sourceTransitArrivalTime - minTransferTime;
+        return OTPFeature.MinimumTransferTimeIsDefinitive.isOn()
+          ? minTransferBoardTime
+          : Math.min(minTransferBoardTime, calcRegularTransferTargetTime.getAsInt());
+      }
```

Comment:
This Return statement and the one below, could possibly be moved outside the if else?
```suggestion
     if (direction.isForward()) {
        int minTransferBoardTime = sourceTransitArrivalTime + minTransferTime;
      } else {
        int minTransferBoardTime = sourceTransitArrivalTime - minTransferTime;
      }
      return OTPFeature.MinimumTransferTimeIsDefinitive.isOn()
          ? minTransferBoardTime
          : Math.max(minTransferBoardTime, calcRegularTransferTargetTime.getAsInt());
```

Reply:
No, the Math calls min/max depending on the search direction.